### DISCUSSION TITLE
Proposed simplified and translingual content category names for WCAG 3 Visual Contrast and APCA-W3

### DISCUSSION THREAD

**Discussion by cyrez** [Myndex/apca-w3#13] [Ideas]
Hello!

This thread is for the discussion of a standard simplified classification of the types of content.

### My Proposal

I would like to propose to use A (A1, A2, A3), B and C to get a universal classification for types of content.

 
### The benefits:
- Simplicity for future usage, to provide a better perception of the content categories.
- Translingual: easier for understanding content types wherever your language.
- Easy to remind.

 
### Content Types
- **A** = **FLUENT**
Fluent Readability
  - **A1** : Block/Body Text
    _Defined as: a block or column of more than two continuous lines of content text._

  - **A2** : Primary content that is not body text
  _Defined as: up to two and a half lines of continuous text._
    1. Includes headlines, captions, and images of text, if they contribute to content
    2. primary navigation and primary menus
    3. asides, tool-tips, spoilers, "continued"
    4. user entered text in forms, fields, etc
    5. does not include text the user controls or adjusts in size or color

  - **A3** : Large Fluent header/title content
  _Defined as: fluent subcategory of "large content" such as big, bold headlines, and generally referring to text larger than 36px._

- **B** = **SUB-FLUENT**
Soft Readability
_Defined as: non-primary content with relaxed readability needs._
  1. Applies to certain aspects of dataviz (call-outs, de-enhanced aspects of visual hierarchy)
  2. Category can include secondary/tertiary nav such as to TOS/EULA
  3. May include byline, short duplicative captions, footer matter.

- **C** = **NON-FLUENT**
Minimum Legibility
  _Defined as: non-content text of an incidental nature._
    1. This category includes non-informative placeholder text, disabled controls, copyright/trademark/license bugs (but not actual license or legal text blocks)
  - Logo or Branding, brand related logo, symbol, service mark.
    1. This category relates to specific colors that are required as part of a brand or logotype.
  - Incidental Text in Images text in images not critical to the understanding, nor specifically contributing to the content.

Note: A3 guideline thresholds similar to B, so maybe a mix possible?

So, a good or a bad idea?

What's your insights?

Cyril

**Comment by Myndex** [comment]
Hi @cyrezdev! Thank you for the submission, we will review in the next Visual Contrast subgroup.

Also, just FYI, the better forum for topics like this is actually [APCA Main Repo Discussions](https://github.com/Myndex/SAPC-APCA/discussions)


### As for abstract labels
My gut reaction is that I myself prefer descriptive key words to prevent:

1) Don't want to be confused with WCAG's AA AAA etc.

2) It is already a difficult to grasp the abstract science here in the first place, so for labels we want more stickyness.

3) I also don't want it to come across as "levels"... we were basing conformance on a 5-level system  at one point, but it became confusing for many.


### Genesis of use cases of text

Developing use cases was in-part a way to develop some rational levels of performance or conformance, based on actual design goals and user's actual needs, and especially for preserving a visual hierarchy.

I discussed this is much greater detail at the main discussion forum in [this thread #75](https://github.com/Myndex/SAPC-APCA/discussions/75#discussioncomment-7017344) And I get into the minutia in [discussion 39]((https://github.com/Myndex/SAPC-APCA/discussions/39) come post in either of those threads, or make any thread of your own if you'd like.

**Comment by cyrez** [comment]
Hi @Myndex !

I didn't expect an answer so fast!

First, i posted here as i was focused on W3 not about the APCA algorithm in itself.

So, sorry if i posted in the wrong section. I will then just answer your message, and post later in the other forum.

### About why i opened this thread:
Not everyone is speaking English. That's why i thought that translingual key words would be more inclusive.

Even if i speak English, i'm not native. For example, i don't know how we can translate "Fluent text" in French with a good understanding for people who will need to use APCA.

Also, "Body-text" for a frontend designer could be confusing (eg. is it all texts included in html <body> markup or the main content of a page included in the <main> markup).

> Don't want to be confused with WCAG's AA AAA etc.

I agree!

But isn't what bronze, silver and gold would be as new accessibility levels?

> It is already a difficult to grasp the abstract science here in the first place, so for labels we want more stickyness.

Exactly the reason to fin good terms/keywords that would make easy to use for all.
Apca is a great tool for color contrast (i'm colorblind and have a corneal dystrophy, and i love the job you did with APCA!).
But it is complex at first sight.

> I also don't want it to come across as "levels"... we were basing conformance on a 5-level system at one point, but it became confusing for many.

Not "levels" but "types of content"

IMO, body-text, fluent text, sub-fluent, non-fluent, spot... are a mixed of levels and categories...

For sure, it's the most difficult part to make something complex easy to understand, and more important, easy to integrate by everyone and not only experts!

I love APCA, and i just want to make it widely use in a near future! 👍 

Personnal side note: i'm building a kind of CSS framework, and i've almost finish a sass library to auto generate accessible colors using APCA agorithm. I didn't check yet all the stuff about licence (i've found the page for it!), but i will share it with you when ready.

> I discussed this is much greater detail at the main discussion forum in https://github.com/Myndex/SAPC-APCA/discussions/75#discussioncomment-7017344 And I get into the minutia in [discussion 39]((https://github.com/Myndex/SAPC-APCA/discussions/39) come post in either of those threads, or make any thread of your own if you'd like.

Already read the first link. Not the second one (with interesting insights!)
Thank you!
